packages/hash/api/src/model/knowledge/link.model.ts

```diff
@@ -111,19 +110,191 @@ export default class {
+   */
+  static async create(
+    graphApi: GraphApi,
+    params: LinkModelCreateParams,
+  ): Promise<LinkModel> {
+    const { sourceEntityModel, linkTypeModel, createdBy } = params;
+    const siblingLinks = await sourceEntityModel.getOutgoingLinks(graphApi, {
+      linkTypeModel,
+    });
+
+    /** @todo: rely on Graph API validation instead of manually checking whether sibling links are ordered */
+    const hasOrderedSiblingLink = siblingLinks[0]?.index !== undefined;
+
+    const index = hasOrderedSiblingLink
+      ? // if siblings are ordered and an index is provided, use the provided index
+        params.index ??
+        // if siblings are ordered and no index is provided, default to the end of the list of links
+        siblingLinks.length
+      : siblingLinks.length === 0
+      ? // if siblings are not ordered because there are no siblings, allow the link to be ordered
+        params.index
+      : // if siblings are not ordered and there are siblings, don't allow the link to be ordered
+        undefined;
+
+    if (index !== undefined) {
+      /** @todo: rely on Graph API to validate the index */
+      if (index < 0 || index > siblingLinks.length) {
+        throw new Error("Provided link index is out of bounds");
+      }
+
+      await Promise.all(
+        siblingLinks
+          .filter((sibling) => sibling.index! >= index)
+          .map((sibling) =>
+            sibling.updateWithoutUpdatingSiblings(graphApi, {
+              updatedIndex: sibling.index! + 1,
+              updatedBy: createdBy,
+            }),
+          ),
+      );
+    }
+
+    return await LinkModel.createLinkWithoutUpdatingSiblings(graphApi, {
+      ...params,
+      index,
+    });
+  }
+
+  private async updateWithoutUpdatingSiblings(
+    graphApi: GraphApi,
+    params: { updatedIndex: number; updatedBy: string },
+  ) {
+    const { updatedIndex, updatedBy } = params;
+
+    const { index: previousIndex } = this;
+
+    if (previousIndex === undefined) {
+      throw new Error("Cannot make an un-ordered link ordered");
+    }
+
+    if (previousIndex === updatedIndex) {
+      throw new Error("No-op: link already has this index");
+    }
+
+    /**
+     * @todo: call dedicated Graph API method to update the index of a link instead of re-creating the link manually
+     * @see https://app.asana.com/0/1202805690238892/1203031430417465/f
+     */
+    await this.removeWithoutUpdatingSiblings(graphApi, {
+      removedBy: updatedBy,
+    });
+
+    const updatedLink = await LinkModel.createLinkWithoutUpdatingSiblings(
+      graphApi,
+      {
+        ...this,
+        index: updatedIndex,
+        createdBy: updatedBy,
+      },
+    );
+
+    return updatedLink;
+  }
+
+  /**
+   * Update the link
+   *
+   * @param params.updatedIndex - the updated index of the link
+   * @param params.updatedBy - the account updating the link
+   */
+  async update(
+    graphApi: GraphApi,
+    params: { updatedIndex: number; updatedBy: string },
```

Comment:
Can we use `updatedById`

Reply:
https://github.com/hashintel/hash/pull/1081/commits/078568b93b61f7b3a0cb85e506f5f08328b89b0c